### ISSUE TITLE
Unknown layer type: L2Normalize

### ISSUE THREAD

**Issue by derkbreeze** [Seth-Park/text_objseg_caffe#3]
Hi Seth:
Thank you for your great work.
When I train the detection model. I got the following error:

I0322 11:15:13.847193 15945 net.cpp:129] Top shape: 50 1000 (50000)
I0322 11:15:13.847195 15945 net.cpp:137] Memory required for data: 5770873800
I0322 11:15:13.847198 15945 layer_factory.hpp:77] Creating layer img_l2norm
F0322 11:15:13.847214 15945 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: L2Normalize (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, Concat, ContrastiveLoss, Convolution, Crop, Data, Deconvolution, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, Input, LRN, LSTM, LSTMUnit, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Parameter, Pooling, Power, Python, RNN, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, WindowData)
*** Check failure stack trace: ***
Aborted (core dumped)

I am using official version of caffe and successfully compiled the caffe with pycaffe and anaconda. and followed your steps.But I still got the error. Is there anything I am missing?
My system is Ubuntu16.04 LTS with Nvidia Cuda8.0 Titan Xp
I appreciate if you can help me.

**Comment by derkbreeze** [comment]
https://github.com/akirafukui/caffe/tree/feature/20160617_cb_softattention
This solved my problem

**Comment by WangLanxiao** [comment]
thanks a lot

**Comment by WangLanxiao** [comment]
> https://github.com/akirafukui/caffe/tree/feature/20160617_cb_softattention
> This solved my problem

what did you do in order to solve this problem

**Comment by derkbreeze** [comment]
> > https://github.com/akirafukui/caffe/tree/feature/20160617_cb_softattention
> > This solved my problem
> 
> what did you do in order to solve this problem

Just git clone that repository and compile that caffe. L2 Normalization layer should be included in that caffe version
